gcsio/src/test/java/com/google/cloud/hadoop/gcsio/FakeWriteChannel.java

```diff
@@ -0,0 +1,71 @@
+/*
+ * Copyright 2023 Google Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.google.cloud.hadoop.gcsio;
+
+import com.google.cloud.RestorableState;
+import com.google.cloud.WriteChannel;
+import java.io.IOException;
+import java.nio.ByteBuffer;
+
+/** FakeWriterChannel which writes only half the passed in byteBuffer capacity at a time. */
+public class FakeWriteChannel implements WriteChannel {
+
+  private boolean open = true;
+
+  private boolean throwExceptionOnWrite = false;
```

Comment:
nit: this could be final if we change default constructor to:
```java
  public FakeWriteChannel() {
    FakeWriteChannel(/* throwExceptionOnWrite= */ false);
  }
```

Reply:
Dne.